metrotranscom/doorway
# feat: add text when there are no matching listings

# Pull Request Template

## Issue Overview

This PR addresses #125 


## Description

- Create a new component (ZeroListingsItem)
- Add the above component to the listings page when there are no listings, above the other cards added in https://github.com/metrotranscom/doorway/pull/115
- Add new text and translations for this component
- Restyle outlined-appearances so the button for this section has the correct colors

What is left - for a future PR
- Add functionality to the button that actually clears all filters

## How Can This Be Tested/Reviewed?

There are two main things that need to be tested: that this component shows when there are no listings, and that it doesn't show when there are matching listings. You can start with the latter since you likely have seed listings in the database already.

- Bring up the public site and go to the listings page.
- At the end of the listings list, there should be no panel saying "No matching listings" and instead it should look like this 
![image](https://github.com/metrotranscom/doorway/assets/67125998/1a1e9cb6-fecf-422b-9a90-7dc6f5f1d19d)

Then we can test the view when there are no listings. For that, I followed this approach to delete the listings from the database:

- Open postgres: `sudo -u postgres psql`
- Connect to the bloom database: `\c bloom`
- Delete the external listings: `DELETE FROM external_listings WHERE TRUE;`
- Delete the listings: `DELETE FROM listings WHERE TRUE;`. This command will fail because it depends on other tables. Delete all entries from those tables recursively until you can delete the listings table. This is around ~8 commands.
- Bring up the public site. There should be no listings shown and you should see a panel saying "No matching listings" followed by 3 info cards.
![image](https://github.com/metrotranscom/doorway/assets/67125998/2f93486f-0500-4c08-961c-9cc2c6360966)


## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
